diff -urN watchdog-5.2.3.org/configure.in watchdog-5.2.3/configure.in
--- watchdog-5.2.3.org/configure.in	2003-05-25 22:32:13.000000000 +0200
+++ watchdog-5.2.3/configure.in	2003-05-25 22:32:22.000000000 +0200
@@ -2,7 +2,7 @@
 AC_INIT(include/extern.h)
 
 MAJOR_VERSION=5
-MINOR_VERSION=2.2
+MINOR_VERSION=2.3
 AM_INIT_AUTOMAKE(watchdog, $MAJOR_VERSION.$MINOR_VERSION)
 
 AC_PREFIX_DEFAULT(/usr)
diff -urN watchdog-5.2.3.org/include/sundries.h watchdog-5.2.3/include/sundries.h
--- watchdog-5.2.3.org/include/sundries.h	2003-05-25 22:32:13.000000000 +0200
+++ watchdog-5.2.3/include/sundries.h	2003-05-25 22:32:22.000000000 +0200
@@ -51,7 +51,7 @@
 #if defined(__GNUC_MINOR__) && __GNUC__ == 2 && __GNUC_MINOR__ >= 5
 void die (int errcode, const char *fmt, ...) __attribute__ ((noreturn));
 #else /* GNUC < 2.5 */
-void volatile die (int errcode, const char *fmt, ...);
+void die (int errcode, const char *fmt, ...);
 #endif /* GNUC < 2.5 */
 #else /* !__GNUC__ */
 void die (int errcode, const char *fmt, ...);
diff -urN watchdog-5.2.3.org/src/quotactl.c watchdog-5.2.3/src/quotactl.c
--- watchdog-5.2.3.org/src/quotactl.c	2003-05-25 22:32:13.000000000 +0200
+++ watchdog-5.2.3/src/quotactl.c	2003-05-25 22:33:30.000000000 +0200
@@ -23,8 +23,9 @@
 #include "config.h"
 #endif
 
-#if defined(__alpha__) || defined(__ia64__)
 #include <errno.h>
+
+#if defined(__alpha__) || defined(__ia64__)
 #include <sys/types.h>
 #include <syscall.h>
 #include <unistd.h>
diff -urN watchdog-5.2.3.org/src/sundries.c watchdog-5.2.3/src/sundries.c
--- watchdog-5.2.3.org/src/sundries.c	2003-05-25 22:32:13.000000000 +0200
+++ watchdog-5.2.3/src/sundries.c	2003-05-25 22:32:22.000000000 +0200
@@ -139,7 +139,7 @@
 }
 
 /* Fatal error.  Print message and exit.  */
-void volatile
+void
 die (int err, const char *fmt, ...) {
      va_list args;
 
diff -urN watchdog-5.2.3.org/src/watchdog.c watchdog-5.2.3/src/watchdog.c
--- watchdog-5.2.3.org/src/watchdog.c	2003-05-25 22:32:13.000000000 +0200
+++ watchdog-5.2.3/src/watchdog.c	2003-05-25 22:32:22.000000000 +0200
@@ -521,7 +521,7 @@
     }
    
     /* make sure we get our own directory in /var/log */
-    if (mkdir ("/var/log/watchdog", 750) && errno != EEXIST) {
+    if (mkdir ("/var/log/watchdog", 0750) && errno != EEXIST) {
 	fprintf(stderr, "%s error:\n", progname);
         fprintf(stderr, "Cannot create directory /var/log/watchdog\n");
 	exit (1);
